src/types.rs

```diff
@@ -321,6 +322,17 @@ pub enum Event {
     },
 }
 
+#[derive(Debug)]
+pub struct WalletTransaction {
+    pub txid: bitcoin::Txid,
+    pub chain_position: ChainPosition<ConfirmationBlockTime>,
+    pub transaction: Arc<bitcoin::Transaction>,
+    pub fee_sats: i64,
+    pub received_sats: i64,
+    pub sent_sats: i64,
```

Comment:
Probably best to use the `Amount` type here?

Reply:
Yes!